internal/deb/extract.go

```diff
@@ -58,8 +59,12 @@ func Extract(pkgReader io.Reader, options *ExtractOptions) (err error) {
 	if err != nil {
 		return err
 	}
+	validOpts := *options
+	if validOpts.Creator == nil {
+		validOpts.Creator = fsutil.NewCreator()
```

Comment:
This does not seem great as it's doing a shallow copy and we have no idea of the consequences. If there's intended mutation going on to the values of the options, they won't be moved out anymore. If on the other hand, the goal is to not mutate, this won't help in cases where the fields are pointers.

There are multiple options here, but the easiest here seems to be simply forcing the call site to provide the creator.  We don't have many call sites anyway, and tests are already doing this in a custom way I believe.

Reply:
I agree with you. I added it because of [your comment](https://github.com/canonical/chisel/pull/105#discussion_r1469389766) about the panic but I agree that, because it is an internal function, we can just add it in each call site.

Reply:
Yeah, it's a bit of a bind because it'd be fine to either 1) Use a default global value that doesn't increase in memory size over time, or 2) Use a local value that is cheap to create. I'm fine with either, but would prefer to not be creating maps, populating maps, and then destroying (GCing) them on every iteration of a loop.  Seems wasteful for no greater good.